src/spec.ts

```diff
@@ -319,9 +339,44 @@ export function alwaysNoOcclusion(spec: ExtendedUnitSpec): boolean {
   return vlEncoding.isAggregate(spec.encoding);
 }
 
-export function fieldDefs(spec: ExtendedUnitSpec): FieldDef[] {
+/* Accumulate non-duplicate fieldDefs in a dictionary */
```

Comment:
Actually you need two.   But make `fieldDefs` one line return that calls `fieldDefIndex` method which basically is the version that returns dictionary (and takes existing `dict: Dict<FieldDef> = {}` as second parameter)